hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieMetadataSync.java

```diff
@@ -0,0 +1,529 @@
+      List<HoodieInstant> pendingInstantsFromLastSync = sourceTableSyncMetadata.get().getInstantsToConsiderForNextSync().stream().map(instant ->
+          sourceActiveTimeline.findInstantsAfterOrEquals(instant, 1).getInstants().stream().findFirst().orElse(null)
+      ).collect(Collectors.toList());
+      pendingInstantsFromLastSync.addAll(newInstantsToSync);
+      return pendingInstantsFromLastSync;
+    } else {
+      return sourceTableMetaClient.reloadActiveTimeline().getInstants();
+    }
+  }
+
+  private HoodieCommitMetadata buildHoodieCommitMetadata(HoodieCommitMetadata sourceCommitMetadata, String commitTime) {
+    HoodieCommitMetadata tgtTableCommitMetadata = new HoodieCommitMetadata();
+
+    for (HoodieWriteStat writeStat : sourceCommitMetadata.getWriteStats()) {
+      String partition = writeStat.getPartitionPath();
+      fixWriteStatusForExternalPaths(writeStat, commitTime);
+      tgtTableCommitMetadata.addWriteStat(partition, writeStat);
+    }
+
+    WriteOperationType operationType = sourceCommitMetadata.getOperationType();
+    tgtTableCommitMetadata.setOperationType(operationType);
+    tgtTableCommitMetadata.setCompacted(false);
+    if (!sourceCommitMetadata.getExtraMetadata().isEmpty()) {
+      sourceCommitMetadata.getExtraMetadata().forEach(tgtTableCommitMetadata::addMetadata);
+    }
+    return tgtTableCommitMetadata;
+  }
+
+  private HoodieReplaceCommitMetadata buildReplaceCommitMetadata(HoodieReplaceCommitMetadata srcReplaceMetadata, String commitTime) {
+    HoodieReplaceCommitMetadata tgtReplaceCommitMetadata = new HoodieReplaceCommitMetadata();
+
+    for (HoodieWriteStat writeStat : srcReplaceMetadata.getWriteStats()) {
+      String partition = writeStat.getPartitionPath();
+      fixWriteStatusForExternalPaths(writeStat, commitTime);
+      tgtReplaceCommitMetadata.addWriteStat(partition, writeStat);
+    }
+
+    WriteOperationType operationType = srcReplaceMetadata.getOperationType();
+    tgtReplaceCommitMetadata.setOperationType(operationType);
+    tgtReplaceCommitMetadata.setCompacted(false);
+    tgtReplaceCommitMetadata.setPartitionToReplaceFileIds(srcReplaceMetadata.getPartitionToReplaceFileIds());
+    if (!srcReplaceMetadata.getExtraMetadata().isEmpty()) {
+      srcReplaceMetadata.getExtraMetadata().forEach(tgtReplaceCommitMetadata::addMetadata);
+    }
+    return tgtReplaceCommitMetadata;
+  }
+
+  private void fixWriteStatusForExternalPaths(HoodieWriteStat writeStat, String commitTime) {
+    String originalPath = writeStat.getPath();
+    writeStat.setPath(ExternalFilePathUtil.appendCommitTimeAndExternalFileMarker(originalPath, commitTime));
+  }
+
+  private HoodieWriteConfig getWriteConfig(
+      Schema schema,
+      HoodieTableMetaClient metaClient,
+      String basePathOverride) {
+    return getWriteConfig(schema, Integer.parseInt(HoodieArchivalConfig.MAX_COMMITS_TO_KEEP.defaultValue()),
```

Comment:
why hard coding to 120?
we should just take configs set while running the spark submit jobs via `--hoodie-conf "a-b"` and feed them into the write config here

Reply:
same for archival config at L 486.